docs/correlating-traces-and-logs.md

```diff
@@ -0,0 +1,132 @@
+# Correlating Trace and Logs
+
+The [OpenTelemetry trace API] can be used to extract trace metadata from a
+context containing a span. That metadata can then be used to annotate log
+events so they will be correlated with the trace.
+
+## Extracting Trace Metadata
+
+To extract trace metadata, use the [`SpanContextFromContext`] function. This
+will extract the trace metadata from an [`context.Context`] and return it in
+the form of a [`SpanContext`].
+
+```go
+spanContext := trace.SpanContextFromContext(ctx)
+if !spanContext.IsValid() {
+	// ctx does not contain a valid span.
+    // There is no trace metadata to add.
+    return
+}
+```
+
+## Annotating Log Events
+
+Once you have a [`SpanContext`], you can use the trace metadata it contains to
+annotate log events. How this is done will depend on the logging library used.
+
+### Structured Logging
+
+If you use a structured logger, the trace metadata should be added as logger
+fields.
+
+For example, using the [zap] logging library:
+
+```go
+logger, _ := zap.NewProduction()
+defer logger.Sync()
+logger = logger.With(
+    zap.String("trace_id", spanContext.TraceID().String()),
+    zap.String("span_id", spanContext.SpanID().String()),
+    zap.String("trace_flags", spanContext.TraceFlags().String()),
+)
+logger.Info("failed to fetch URL", zap.String("URL", url))
+```
+
+### Unstructured Logging
+
+If using unstructured logging, the trace metadata can be added as a part of the
+logged message.
+
+For example using the standard library [`log`] package:
+
+```go
+log.Printf(
+    "(trace_id: %s, span_id: %s, trace_flags: %s): failed to fetch URL: %s",
+    spanContext.TraceID().String(),
+    spanContext.SpanID().String(),
+    spanContext.TraceFlags().String(),
```

Comment:
It'd be nice to state what metadata is available at the beginning of this document (`TraceID`, `SpanID`, etc.)